src/components/Editor/Runners/PyodideRunner/PyodideRunner.jsx

```diff
@@ -0,0 +1,364 @@
+    pyodideWorker.postMessage({ method: "writeFile", filename, content });
+  };
+
+  const inputSpan = () => {
+    const span = document.createElement("span");
+    span.setAttribute("id", "input");
+    span.setAttribute("spellCheck", "false");
+    span.setAttribute("class", "pythonrunner-input");
+    span.setAttribute("contentEditable", "true");
+    return span;
+  };
+
+  const getInputElement = () => {
+    const pageInput = document.getElementById("input");
+    const webComponentInput = document.querySelector("editor-wc")
+      ? document.querySelector("editor-wc").shadowRoot.getElementById("input")
+      : null;
+    return pageInput || webComponentInput;
+  };
+
+  const getInputContent = async (element) => {
+    element.focus();
+
+    return new Promise(function (resolve, reject) {
+      element.addEventListener("keydown", function removeInput(e) {
+        const ctrlD = e.ctrlKey && e.key.toLowerCase() === "d";
+        const lineEnded = e.key === "Enter" || ctrlD;
+
+        if (lineEnded) {
+          element.removeEventListener(e.type, removeInput);
+          const content = element.innerText;
+          element.removeAttribute("id");
+          element.removeAttribute("contentEditable");
+          element.innerText = content + "\n";
+
+          document.addEventListener("keyup", function storeInput(e) {
+            if (lineEnded) {
+              document.removeEventListener(e.type, storeInput);
+              resolve({ content, ctrlD });
+            }
+          });
+        }
+      });
+    });
+  };
+
+  const shiftFocusToInput = (event) => {
+    if (document.getSelection().toString().length > 0) {
+      return;
+    }
+
+    const inputBox = getInputElement();
+    if (inputBox && event.target !== inputBox) {
+      const input = getInputElement();
+      const selection = window.getSelection();
+      selection.removeAllRanges();
+
+      if (input.innerText && input.innerText.length > 0) {
+        const range = document.createRange();
+        range.setStart(input, 1);
+        range.collapse(true);
+        selection.addRange(range);
+      }
+      input.focus();
+    }
+  };
+
+  const disableInput = () => {
+    const element = getInputElement();
+
+    if (element) {
+      element.removeAttribute("id");
+      element.removeAttribute("contentEditable");
+    }
+  };
+
+  return (
+    <div className={`pythonrunner-container`}>
+      {isSplitView ? (
+        <>
+          {hasVisual ? (
+            <div className="output-panel output-panel--visual">
+              <Tabs forceRenderTabPanel={true}>
+                <div className="react-tabs__tab-container">
+                  <TabList>
+                    <Tab key={0}>
+                      <span className="react-tabs__tab-text">
+                        {t("output.visualOutput")}
+                      </span>
+                    </Tab>
+                  </TabList>
+                  {!isEmbedded && hasVisual ? <OutputViewToggle /> : null}
```

Comment:
⛏️  happens in a couple of places but a bunch of the ternaries can be simplified. Minor though
```suggestion
                  {!isEmbedded && hasVisual && <OutputViewToggle />}
```